libs/stm32f0_drivers/usart.c

```diff
@@ -0,0 +1,295 @@
+#include <stdio.h>
+#include <errno.h>
+
+#include <FreeRTOS.h>
+#include <timers.h>
+#include <task.h>
+#include <queue.h>
+#include <semphr.h>
+
+#include <stm32f0xx_usart.h>
+#include <stm32f0xx_misc.h>
+#include <stm32f0xx_gpio.h>
+#include <stm32f0xx_rcc.h>
+
+#include <usart.h>
+
+struct usart {
+	USART_TypeDef *periph;
+	bool autocrlf;
+	bool initialized;
+
+	struct {
+		xQueueHandle tx;
+		xQueueHandle rx;
+	} buffers;
+
+	struct {
+		void (*periph_clock_cmd)(uint32_t, FunctionalState);
+		uint32_t rccperiph;
+	} rcc;
+
+	struct {
+		GPIO_TypeDef *rxport;
+		uint16_t rxpin;
+		void (*rx_clock_cmd)(uint32_t, FunctionalState);
+		uint32_t rx_pin_clock;
+		uint8_t         rx_pinsource;
+		GPIO_TypeDef *txport;
+		uint16_t txpin;
+		void (*tx_clock_cmd)(uint32_t, FunctionalState);
+		uint32_t tx_pin_clock;
+		uint8_t         tx_pinsource;
+	} gpios;
+
+	bool rx_overflow;
```

Comment:
It's a watermark for debugging. Not much to be done (though I suppose I can add a method to check this if the situation were to arise when that info were useful)